src/util/std_expr.h

```diff
@@ -4283,47 +4283,81 @@ class infinity_exprt : public nullary_exprt
 };
 
 /// \brief A let expression
-class let_exprt : public ternary_exprt
+class let_exprt : public binary_exprt
 {
 public:
   let_exprt(symbol_exprt symbol, exprt value, const exprt &where)
-    : ternary_exprt(
+    : binary_exprt(
+        multi_ary_exprt(
+          ID_tuple,
+          {std::move(symbol), std::move(value)},
+          typet(ID_tuple)),
+        ID_let,
+        where,
+        where.type())
+  {
+  }
+
+  using symbol_value_sequencet = std::vector<exprt>;
+
+  let_exprt(symbol_value_sequencet symbol_value_sequence, const exprt &where)
+    : binary_exprt(
+        multi_ary_exprt(
+          ID_tuple,
+          std::move(symbol_value_sequence),
+          typet(ID_tuple)),
         ID_let,
-        std::move(symbol),
-        std::move(value),
         where,
         where.type())
   {
   }
 
+  symbol_value_sequencet &symbol_value_sequence()
+  {
+    return (symbol_value_sequencet &)op0().operands();
```

Comment:
I do find these type casts scary. Right now, this one actually is unnecessary. But if in future there were to be a difference then only runtime crashes would tell us.